### PR TITLE
[eas-cli] Extract entity utilities from update command class

### PR DESCRIPTION
**Pull request by byCedric**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

Continuation of PR #1473 and ENG-5926.

# How

- Moved ensure methods from `UpdatePublish` to `src/branch|channel/queries.ts`.
- Copied other methods from commands `BranchCreate` and `ChannelCreate`, to keep queries contained in `src/branch|channel/queries.ts` files.
- Validated the logging is similar (still slightly different)
  - _branch logging_
    - removed: `Created branch: <branchName>`
    - why: The branch is both listed in the formatted fields (first field), and the "Channel is pointing to branch" logging, seems not worth it to determine if the branch was created or already existed. (_but if people disagree, I'd be happy to change it_)
  - _channel logging_  
    - from: `Created a channel: <channelName> pointed at branch: <branchName>` (_only logged when created_)
    - to: `Channel: <channelName> pointed at branch: <branchName>` (_always logged_) 
    - why: The channel name is not part of the formatted fields at the end. We either need to display this, or maybe even add it to the formatted fields as item? (_instead of the `Log.withTick`_)

# Test Plan

- `eas update --branch new-branch`
- `eas update --branch existing-branch`

### CODE REVIEW

**Comment by byCedric** [comment]
Open question to everyone here:
  1. Do we want to keep exporting these `create<Entity>` or `ensure<Entity>...` methods from other command classes? I don't really like it, since it implicitly creates depending command classes. But, not sure if there is a reason to do it.
  2. Why do we have `src/graphql/queries|mutations` and separate `src/branch|channel/queries.ts` files? Seems like they would be more in place under the graphql client, like most of the non-updates-related entities?
  3. Seeing some inconsistencies with the return types and logic of `ensure<Entity>` methods. Some won't return anything, others just the entity ID. The `ensureChannelExists` also just yolos it and uses potential errors to determine if the channel exists, makes sense in terms of limiting the graphql calls, but it does differ quite a lot compared to `ensureBranchExists`.

File: packages/eas-cli/src/commands/branch/create.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ import { getDisplayNameForProjectIdAsync } from '../../project/projectUtils';
 import { promptAsync } from '../../prompts';
 import { enableJsonOutput, printJsonOnlyOutput } from '../../utils/json';
 
+// NOTE(cedric): copied to src/branch/queries.ts to reuse in multiple commands
```
Comment (wschurman) [inline packages/eas-cli/src/commands/branch/create.ts:19]: can these methods be removed now that they've been factored out?

**Review by wschurman — APPROVED** [review]

File: packages/eas-cli/src/commands/branch/create.ts (hunk 1 above)
Comment (byCedric) [reply packages/eas-cli/src/commands/branch/create.ts:19]: Yeah, pending my open questions. I left a comment so I could do a quick search&destroy operation 😄

Comment (byCedric) [reply packages/eas-cli/src/commands/branch/create.ts]: Done!

**Pull request merged**